src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -170,11 +171,17 @@ private static FileStream CreateSharedBackingObject(Interop.Sys.MemoryMappedProt
                 ?? CreateSharedBackingObjectUsingFile(protections, capacity, inheritability);
         }
 
-        private static FileStream? CreateSharedBackingObjectUsingMemory(
+        private static unsafe FileStream? CreateSharedBackingObjectUsingMemory(
            Interop.Sys.MemoryMappedProtections protections, long capacity, HandleInheritability inheritability)
         {
-            // The POSIX shared memory object name must begin with '/'.  After that we just want something short and unique.
-            string mapName = string.Create(null, stackalloc char[128], $"/corefx_map_{Guid.NewGuid():N}");
+            const int MaxSharedMemoryObjectNameLength = 32; // SHM_NAME_MAX on OSX ARM64, on other systems it's equal PATH_MAX (250)
+            // The POSIX shared memory object name must begin with '/'.  After that we just want something short (32) and unique.
+            Span<char> characters = stackalloc char[MaxSharedMemoryObjectNameLength];
+            char format = 'N';
+            bool success = Guid.NewGuid().TryFormat(characters, out int charsWritten, new ReadOnlySpan<char>(&format, 1));
+            Debug.Assert(success && charsWritten == MaxSharedMemoryObjectNameLength);
+            characters[0] = '/';
+            string mapName = new string(characters);
```

Comment:
Why did you remove use of string.Create?

Reply:
Ah, because Guid.NewGuid().ToString("N").Length == 32 and you need the slash first?

Reply:
In that case, you can format directly into the string, e.g.
```C#
string mapName = string.Create(MaxSharedMemoryObjectNameLength, 0, (span, state) =>
{
    Guid.NewGuid().TryFormat(span, out int charsWritten, "N");
    Debug.Assert(charsWritten == MaxSharedMemoryObjectNameLength);
    span[0] = '/';
});
```